sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/OperatorOverloadsShouldHaveNamedAlternatives.cs

```diff
@@ -0,0 +1,130 @@
+            ["op_Multiply"] = "Multiply",
+            ["op_OnesComplement"] = "OnesComplement",
+            ["op_Subtraction"] = "Subtract",
+            ["op_UnaryNegation"] = "Negate",
+            ["op_UnaryPlus"] = "Plus"
+        };
+
+        private static readonly Dictionary<string, string> operatorNames = new Dictionary<string, string>
+        {
+            ["op_Addition"] = "+",
+            ["op_BitwiseAnd"] = "&",
+            ["op_BitwiseOr"] = "|",
+            ["op_Division"] = "/",
+            ["op_ExclusiveOr"] = "^",
+            ["op_Equality"] = "==",
+            ["op_Inequality"] = "!=",
+            ["op_GreaterThan"] = ">",
+            ["op_LessThan"] = "<",
+            ["op_GreaterThanOrEqual"] = ">=",
+            ["op_LessThanOrEqual"] = "<=",
+            ["op_Decrement"] = "--",
+            ["op_Increment"] = "++",
+            ["op_LeftShift"] = "<<",
+            ["op_RightShift"] = ">>",
+            ["op_LogicalNot"] = "!",
+            ["op_Modulus"] = "%",
+            ["op_Multiply"] = "*",
+            ["op_OnesComplement"] = "~",
+            ["op_Subtraction"] = "-",
+            ["op_UnaryNegation"] = "-",
+            ["op_UnaryPlus"] = "+"
+        };
+
+        private static readonly DiagnosticDescriptor rule =
+            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(rule);
+
+        protected override void Initialize(SonarAnalysisContext context)
+        {
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
+            {
+                var operatorDeclaration = (OperatorDeclarationSyntax)c.Node;
+                var operatorSymbol = c.SemanticModel.GetDeclaredSymbol(operatorDeclaration);
+                if (operatorSymbol == null ||
+                    operatorSymbol.MethodKind != MethodKind.UserDefinedOperator)
+                {
+                    return;
+                }
+
+                var operatorAlternativeMethodName = operatorAlternatives.GetValueOrDefault(operatorSymbol.Name);
+                var operatorName = operatorNames.GetValueOrDefault(operatorSymbol.Name);
+                if (operatorName != null && 
+                    !HasAlternativeMethod(operatorSymbol, operatorAlternativeMethodName))
+                {
+                    c.ReportDiagnostic(Diagnostic.Create(rule, operatorDeclaration.OperatorToken.GetLocation(),
+                        operatorAlternativeMethodName, operatorName));
+                }
+            },
+            SyntaxKind.OperatorDeclaration);
+        }
+
+        private static bool HasAlternativeMethod(IMethodSymbol operatorMethod, string operatorAlternativeMethodName)
+        {
+            return operatorAlternativeMethodName == null ||
+                operatorMethod.ContainingType
+                    .GetMembers(operatorAlternativeMethodName)

```

Comment:
Initially I checked for parameters and return type, but then I saw FxCop does not check that and simplified the code...